specification/_types/query_dsl/specialized.ts

```diff
@@ -121,9 +121,19 @@ export class PercolateQuery extends QueryBase {
   version?: VersionNumber
 }
 
+/**
+ * @variants container
+ */
 export class PinnedQuery extends QueryBase {
-  ids: Id[]
+  /** @variant container_property */
   organic: QueryContainer
+  ids?: Id[]
+  docs?: PinnedDoc[]
+}
```

Comment:
@swallez is this valid? Or should I manually declare the `QueryBase` properties as `container_property` here?

Reply:
`PinnedQuery` isn't a container as it doesn't have variants. Its `organic` property is just a `QueryContainer` (which is indeed a container).

So no, it's not valid, these jstags should not be there and aren't needed.

Here's a breakdown of the structure:
```json5
{
  "query": {                       // query: QueryContainer
    "pinned": {                    //   variant identifier (PinnedQuery)
      "ids": [ "1", "4", "100" ],  //     PinnedQuery field (Id[])
      "organic": {                 //     PinnedQuery field (QueryContainer)
        "match": {                 //       variant identifier (SingleKeyDictionary<Field, MatchQuery>)
          "description": "iphone"  //         SingleKeyDictionary<Field, MatchQuery> entry
        }
      }
    }
  }
}
```